modules/tensor_mechanics/test/tests/czm/czm_large_deformation.i

```diff
@@ -0,0 +1,355 @@
+    axis_origin = '0. 0. 0.'
+    axis_direction = '0. 1. 0.'
+    component = 0
+    variable = disp_x
+    angular_velocity = true
+  [../]
+  [./rotate_y]
+    type = DisplacementAboutAxis
+    boundary = 'front back'
+    function = '90.'
+    angle_units = degrees
+    axis_origin = '0. 0. 0.'
+    axis_direction = '0. 1. 0.'
+    component = 1
+    variable = disp_y
+    angular_velocity = true
+  [../]
+  [./rotate_z]
+    type = DisplacementAboutAxis
+    boundary = 'front back'
+    function = '90.'
+    angle_units = degrees
+    axis_origin = '0. 0. 0.'
+    axis_direction = '0. 1. 0.'
+    component = 2
+    variable = disp_z
+    angular_velocity = true
+  [../]
+[]
+
+[AuxVariables]
+  [./da_dA]
+    family = MONOMIAL
+    order = CONSTANT
+  []
+  [./tczm_X]
+    family = MONOMIAL
+    order = CONSTANT
+  []
+  [./tczm_Y]
+    family = MONOMIAL
+    order = CONSTANT
+  []
+  [./tczm_Z]
+    family = MONOMIAL
+    order = CONSTANT
+  []
+  [./TPK1czm_X]
+    family = MONOMIAL
+    order = CONSTANT
+  []
+  [./TPK1czm_Y]
+    family = MONOMIAL
+    order = CONSTANT
+  []
+  [./TPK1czm_Z]
+    family = MONOMIAL
+    order = CONSTANT
+  []
+[]
+#
+[AuxKernels]
+  [./da_dA]
+    type = MaterialRealAux
+    boundary = 'interface'
+    property = da_dA_avg
+    execute_on = 'TIMESTEP_END'
+    variable = da_dA
+  []
+  [./tczm_X]
+    type = MaterialRealVectorValueAux
+    boundary = 'interface'
+    property = traction_global
+    component = 0
+    execute_on = 'TIMESTEP_END'
+    variable = tczm_X
+  []
+  [./tczm_Y]
+    type = MaterialRealVectorValueAux
+    boundary = 'interface'
+    property = traction_global
+    component = 1
+    execute_on = 'TIMESTEP_END'
+    variable = tczm_Y
+  []
+  [./tczm_Z]
+    type = MaterialRealVectorValueAux
+    boundary = 'interface'
+    property = traction_global
+    component = 2
+    execute_on = 'TIMESTEP_END'
+    variable = tczm_Z
+  []
+  [./TPK1czm_X]
+    type = MaterialRealVectorValueAux
+    boundary = 'interface'
+    property = PK1traction
+    component = 0
+    execute_on = 'TIMESTEP_END'
+    variable = TPK1czm_X
+  []
+  [./TPK1czm_Y]
+    type = MaterialRealVectorValueAux
+    boundary = 'interface'
+    property = PK1traction
+    component = 1
+    execute_on = 'TIMESTEP_END'
+    variable = TPK1czm_Y
+  []
+  [./TPK1czm_Z]
+    type = MaterialRealVectorValueAux
+    boundary = 'interface'
+    property = PK1traction
+    component = 2
+    execute_on = 'TIMESTEP_END'
+    variable = TPK1czm_Z
+  []
+[]
+
+
+[Modules/TensorMechanics/CohesiveZoneMaster]
+  [./czm_ik]
+    boundary = 'interface'
+    incremental_material=true
+  [../]
+[]
+
+[Controls]
+  [./c1]
+    type = TimePeriod
+    enable_objects = 'BCs::fix_x BCs::fix_y BCs::fix_z BCs::back_z'
+    disable_objects = 'BCs::rotate_x BCs::rotate_y BCs::rotate_z'
+    start_time = '0'
+    end_time = '1'
+  [../]
+[]
+
+[Modules]
+  [./TensorMechanics]
+    [./Master]
+      [./all]
+        strain = FINITE
```

Comment:
Do all those combination work?
1. bulk: small total formulation, interface: small or large
2. bulk: small incremental formulation, interface: small or large
3. bulk: finite strain, interface, small or large

Reply:
yes, all of them work. 
But you should be careful about what you do, as you are using different stresses for imposing equilibrium
The interface small formulation should be used together with a small bulk formulation (incremental or total), 
The interface large formulation should be used together with a finite strain formulation. 
you should not use the large interface model with a small bulk formulation.

Reply:
Could you add those information to the documentation? Thanks.